### PR TITLE
Modification units poliastro_plotting_core.py

### LINKED ISSUE #1291: Change plotting units
💡 **Feature request**

By default, plots are shown with the axis in kilometers, like this one:

plot

And the reason is that units are hardcoded here:

https://github.com/poliastro/poliastro/blob/19880062af49dbf92147528f1b98de378a3d4e89/src/poliastro/plotting/core.py#L186-L190

It would be nice to be able to change those units easily. This can be manually done manipulating the Plotly figure, but it's far from ideal.

A possible solution would be to add a `unit` parameter to the initializer of `_PlotlyOrbitPlotter`, here:

https://github.com/poliastro/poliastro/blob/19880062af49dbf92147528f1b98de378a3d4e89/src/poliastro/plotting/core.py#L17-L18

so that in `_plot_coordinates` we end up with this code:

```python
    def _plot_coordinates(self, coordinates, label, colors, dashed):
        trace = Scatter3d(
            x=coordinates.x.to_value(self.unit),
            y=coordinates.y.to_value(self.unit),
            z=coordinates.z.to_value(self.unit),
```

Notice that this would require modifying also the initializers of `OrbitPlotter3D` and `OrbitPlotter2D`, as well as the corresponding `super` calls.

### PR DESCRIPTION
**Pull request by JohaMR**
fix #1291

### CODE REVIEW

**Review by astrojuanlu — COMMENTED** [review]
Congratulations for your first pull request 🎉 

All the tests have passed, which means that nothing broke 👍🏼

Only the quality check failed, because there are some whitespace things to change. No rush.

Before we continue, we are going to try to see if this solves your original request. I will give you more instructions soon.

**Comment by astrojuanlu** [comment]
@JohaMR To see if this solves your original request, can you please install this version of poliastro:

```
pip install https://github.com/JohaMR/poliastro/archive/refs/heads/patch-1.zip
```

and see if you can now change the units of your plot?

```python
from astropy import units as u
from astropy.time import Time
from poliastro.twobody import Orbit
from poliastro.plotting import OrbitPlotter3D

ceres = Orbit.from_sbdb("510466")
epoch = Time("2021-05-30 14:10:00", scale="tdb")
ceres_may = ceres.propagate(epoch)
ceres_may.epoch.iso
frame = OrbitPlotter3D(unit=u.au)  # <-------------------------
frame.plot(ceres_may, label="510466")
frame.show()
```

? If it works, we can address the whitespace changes and merge this 🚀

**Comment by JohaMR** [comment]
> @JohaMR Para ver si esto resuelve su solicitud original, puede instalar esta versión de poliastro:
> 
> ```
> pip install https://github.com/JohaMR/poliastro/archive/refs/heads/patch-1.zip
> ```
> 
> y vea si ahora puede cambiar las unidades de su parcela?
> 
> ```python
> de  unidades de importación de astropía  como u de astropía . time import Time from poliastro . importación de dos cuerpos Orbit de poliastro . trazado de importación OrbitPlotter3D ceres = Orbit . from_sbdb ( "510466" )
>  epoch = Time ( "2021-05-30 14:10:00" , scale = "tdb" )
>  ceres_may = ceres . Propagar (   
>    
>    
>    
> 
>       época )
>  ceres_may . época . marco iso 
> = OrbitPlotter3D ( unidad = u . au )   # <------------------------- marco . trama ( ceres_may , label = "510466" )
>  marco . mostrar ()  
> ```
> 
> ? Si funciona, podemos abordar los cambios de espacios en blanco y fusionar esto🚀

**Comment by JohaMR** [comment]
I did what you suggest and it didn't work

**Comment by astrojuanlu** [comment]
Thanks! No need to close the pull request, we can keep making changes

**Comment by astrojuanlu** [comment]
There's at least a couple more things we have to change:

https://github.com/poliastro/poliastro/blob/48c854a4ad5500116f7c09ca171f77a60e2de04b/src/poliastro/plotting/core.py#L152-L155

https://github.com/poliastro/poliastro/blob/48c854a4ad5500116f7c09ca171f77a60e2de04b/src/poliastro/plotting/core.py#L251-L254

Instead of the string `"(km)"`, we should use `f"({self._unit})".

_Still_, the unit seems to have no effect on the values, and I don't understand why. I will keep investigating.

**Comment by astrojuanlu** [comment]
There are more parts of the plotting code that default to kilometers. This is more complicated than I thought 😃 But we'll do it!

**Comment by astrojuanlu** [comment]
@JohaMR could you:

1. Make the edits I suggested in https://github.com/poliastro/poliastro/pull/1318#issuecomment-917384373
2. Add `unit=unit` in the `super().__init__` calls of `OrbitPlotter3D` and `OrbitPlotter2D`,
3. Repeat `pip install https://github.com/JohaMR/poliastro/archive/refs/heads/patch-1.zip`, and
4. Check if now the units are shown?

File: src/poliastro/plotting/core.py [hunk 1]
```diff
@@ -145,14 +147,14 @@ def show(self):
...
-        super().__init__(figure, num_points=num_points, plane=plane)
+    def __init__(self, figure=None, dark=False, *, num_points=150, plane=None, unit=u.km):
+        super().__init__(figure, num_points=num_points, plane=plane, unit=unit)
         self._layout = Layout(
             autosize=True,
             scene=dict(
-                xaxis=dict(title="x (km)"),
-                yaxis=dict(title="y (km)"),
-                zaxis=dict(title="z (km)"),
+                xaxis=dict(title="x f"({self._unit})""),
```
Comment (astrojuanlu) [inline src/poliastro/plotting/core.py:155]: This is a syntax error, it should be:

```suggestion
                xaxis=dict(title=f"x ({self._unit})"),
```

and so forth

**Review by astrojuanlu — CHANGES_REQUESTED** [review]

**Comment by astrojuanlu** [comment]
First checkbox done ✅  missing the other 3 :)

**Comment by JohaMR** [comment]
correct code, now it is possible to change units 🎉

**Review by astrojuanlu — APPROVED** [review]
Congratulations! 🎉 

I just made some whitespace fixes to your pull request, and as soon as the Continuous Integration checks pass, we will merge it.

**Pull request merged**